rust/connlib/clients/android/src/lib.rs

```diff
@@ -78,22 +85,27 @@ fn call_method(
 }
 
 fn init_logging(log_dir: PathBuf) {
+    // On Android, we can only init logging once after the lib is loaded. However,
+    // session connect and disconnect can happen many times in that timespan.
+    if LOGGING_GUARD.get().is_some() {
+        return;
+    }
+
+    #[cfg(debug_assertions)]
+    let log_level = LevelFilter::Debug;
+
+    #[cfg(not(debug_assertions))]
+    let log_level = LevelFilter::Info;
+
     // Initializes integration with Logcat for Android
-    // This can be called many times, but will only initialize logging once
-    android_logger::init_once(
-        android_logger::Config::default()
-            .with_max_level(if cfg!(debug_assertions) {
-                LevelFilter::Debug
-            } else {
-                LevelFilter::Info
-            })
-            .with_tag("connlib"),
-    );
-
-    let (file_layer, _guard) = file_logger::layer(log_dir);
-
-    // Calling init twice causes a panic; instead use try_init which will fail
-    // gracefully if this is called more than once.
+    android_logger::init_once(android_logger::Config::default().with_max_level(log_level));
```

Comment:
This might be a bit late but have you looked into https://docs.rs/tracing-android?

I know that tracing subscribers can be initialized more often / unset again. The problem is usually `log` because that one does not allow unsetting. Still, using as many "tracing-native" components might make sense / things a bit easier.

But this can also be left for later if it is working now :)

Reply:
Yeah we did stumble across it but didn't test it since the repo seems a bit inactive. That could be because the Android NDK logging API is fairly mature now. Should be an easy change, I'll give it a shot.

Reply:
Seems to work ok! It's such a small crate we could maintain our own version of it if needed. Worth keeping everything in our tracing setup.

Reply:
> Yeah we did stumble across it but didn't test it since the repo seems a bit inactive. That could be because the Android NDK logging API is fairly mature now. Should be an easy change, I'll give it a shot.

I know the authors from `rust-libp2p` contributions. They are good devs and all the dependencies seem up to date so that is probably just it why there is low activity :)